lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/common/VortexAvroUtils.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.common;
+
+import org.apache.avro.io.*;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.reef.vortex.api.VortexFunction;
+import org.apache.reef.vortex.common.avro.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+/**
+ * Serialize and deserialize Vortex message to/from byte array.
+ */
+public final class VortexAvroUtils {
+  /**
+   * Serialize VortexRequest to byte array.
+   * @param vortexRequest Vortex request message to serialize.
+   * @return Serialized byte array.
+   */
+  public static byte[] toBytes(final VortexRequest vortexRequest) {
+    // Convert VortexRequest message to Avro message.
+    final AvroVortexRequest avroVortexRequest;
+    switch (vortexRequest.getType()) {
+    case ExecuteTasklet:
+      final TaskletExecutionRequest taskletExecutionRequest = (TaskletExecutionRequest) vortexRequest;
+      // TODO[REEF-1005]: Allow custom codecs for input/output data in Vortex.
+      final byte[] serializedInput = SerializationUtils.serialize(taskletExecutionRequest.getInput());
+      // TODO[REEF-1003]: Use reflection instead of serialization when launching VortexFunction
```

Comment:
Do you want me to elaborate the TODO comment instead of having just JIRA title? Or do you want me to give some comments of why do we need those TODOs?